client/blocks/importer/hooks/use-site-can-migrate.ts

```diff
@@ -35,29 +35,34 @@ export function useSiteMigrateInfo(
 		);
 	};
 
-	const onMigrationEnabledSuccess = ( data: MigrationEnabledResponse ) => {
-		const canMigrate = checkCanSiteMigrate( data );
-		if ( canMigrate ) {
-			setSiteCanMigrate( true );
-			setSourceSiteId( data.source_blog_id );
-		} else {
-			resetMigrationEnabled();
+	const {
+		refetch,
+		isFetching: isMigrationEnabledFetching,
+		isError,
+		isSuccess,
+		data,
+	} = useMigrationEnabledInfoQuery( targetSiteId, sourceSiteSlug, fetchMigrationEnabledOnMount );
+
+	useEffect( () => {
+		if ( isSuccess && data ) {
+			const canMigrate = checkCanSiteMigrate( data );
+			if ( canMigrate ) {
+				setSiteCanMigrate( true );
+				setSourceSiteId( data.source_blog_id );
+			} else {
+				resetMigrationEnabled();
+			}
+			onfetchCallback?.( canMigrate );
 		}
-		onfetchCallback && onfetchCallback( canMigrate );
-	};
+	}, [ data, isSuccess, onfetchCallback ] );
 
-	const onMigrationEnabledError = () => {
-		resetMigrationEnabled();
-		const canMigrate = false;
-		onfetchCallback && onfetchCallback( canMigrate );
-	};
-	const { refetch, isFetching: isMigrationEnabledFetching } = useMigrationEnabledInfoQuery(
-		targetSiteId,
-		sourceSiteSlug,
-		fetchMigrationEnabledOnMount,
-		onMigrationEnabledSuccess,
-		onMigrationEnabledError
-	);
+	useEffect( () => {
+		if ( isError ) {
+			resetMigrationEnabled();
+			const canMigrate = false;
+			onfetchCallback && onfetchCallback( canMigrate );
```

Comment:
We could potentially change this to use optional chaining just like you changed it above:

```suggestion
			onfetchCallback?.( canMigrate );
```

Reply:
I removed this callback in https://github.com/Automattic/wp-calypso/pull/80960/commits/6150544ea20e7ea0d4279f72c462eae5959e31a1